core-bundle/src/Migration/Version505/FormStoreSessionMigration.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Migration\Version505;
+
+use Contao\CoreBundle\Migration\AbstractMigration;
+use Contao\CoreBundle\Migration\MigrationResult;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Types\Types;
+
+/**
+ * @internal
+ */
+class FormStoreSessionMigration extends AbstractMigration
+{
+    public function __construct(private readonly Connection $connection)
+    {
+    }
+
+    public function shouldRun(): bool
+    {
+        $schemaManager = $this->connection->createSchemaManager();
+
+        if (!$schemaManager->tablesExist('tl_form')) {
+            return false;
+        }
+
+        $columns = $schemaManager->listTableColumns('tl_form');
+
+        if (isset($columns['storesession'])) {
+            return false;
+        }
+
+        return (bool) $this->connection->fetchOne('SELECT TRUE FROM tl_form LIMIT 1');
+    }
+
+    public function run(): MigrationResult
+    {
+        $platform = $this->connection->getDatabasePlatform();
+        $schemaManager = $this->connection->createSchemaManager();
+        $fromTable = $schemaManager->introspectSchema()->getTable('tl_form');
+        $toTable = clone $fromTable;
+        $toTable->addColumn('storeSession', Types::BOOLEAN, ['default' => false]);
+
+        $tableDiff = $schemaManager->createComparator()->compareTables($fromTable, $toTable);
+
+        foreach ($platform->getAlterTableSQL($tableDiff) as $query) {
+            $this->connection->executeQuery($query);
+        }
```

Comment:
We should only run the specific `ALTER TABLE` query that is required for the migration. Any other changes should be shown to the user on the CLI so they can decide whether or not they want to apply them.

Reply:
That will not work in this case.

1. We want to leave the setting _disabled_ by default.
2. For _existing_ forms we want to _enable_ the setting.
3. For the migration to know when to run it determines whether the `storeSession` field has already been added or not.
4. If the field was not added, it will add the field and set it to _enabled_.
5. If the field was already added, the migration will not run.

Reply:
All of that is already implemented. You just have to replace lines 48 to 58 with a simple `ALTER TABLE …` statement as we do in other migrations:

https://github.com/contao/contao/blob/39e7051fdaecb066327f05df88377e147618aa02/core-bundle/src/Migration/Version503/FrontendModulesMigration.php#L48-L53

Reply:
The code you referenced runs this `ALTER TABLE`, nothing else. I deliberately did not use a direct statement due to issue like https://github.com/contao/contao/issues/7985.

Reply:
Then I cannot merge the PR, I'm afraid. A concept change requires previous discussion and must be implemented for all migrations, not just for one.

Reply:
But we already used it in #8165. What is different here?

Reply:
I don't see `compareTables()` in #8165?

Reply:
#8165 did not need `compareTables()` as it creates a _new_ table using Doctrine DBAL. This PR on the other hand adds a new column of an existing table using Doctrine DBAL.

Reply:
I would agree with @leofeyer here. A simple `ALTER TABLE` statement should always work, we use that all over the place (I also just used this in https://github.com/contao/contao/pull/8365)

Reply:
Personally I do not want to look up / remember what statement I need to execute when I need to create a column with the configuration `['type' => 'boolean', 'default' => false]`. Ideally the database abstraction layer should take care of this.

Reply:
Personally, I will need to look up what the DBAL abstraction notation is, I don't need to look up SQL for that 😂

Reply:
Using the DBAL abstraction it will always automatically use the correct type. Imagine if MySQL introduces a native boolean type and DBAL implement this for that platform. If we don't use the abstraction this migration will execute a different statement then our DCA migration will do, since the DCA uses the abstraction and this migration would not.